raft/src/main/java/org/apache/kafka/snapshot/SnapshotReader.java

```diff
@@ -77,6 +79,18 @@ public int lastContainedLogEpoch() {
         return snapshotId.epoch;
     }
 
+    /**
+     * Returns the timestamp of the last log offset which is represented in the snapshot.
+     */
+    public long lastContainedLogTimestamp() {
+        if (!lastContainedLogTimestamp.isPresent()) {
+            // nextBatch is expected to be empty
+            nextBatch = nextBatch();
```

Comment:
Is the whole snapshot written as a single batch? Otherwise, it seems like if you open a snapshot for reading and call this method, you'll just get the append time of the first batch. 

Or is the idea that this method is called only after the reader has read through all the batches?

Reply:
I documented this in the PR description and comment in this file: https://github.com/apache/kafka/pull/10946/files/1964e026b783235e2d9af9c0b116f15ed31e0a5b#diff-83295dbf4af9755c79987b390f72f53dda47af9f82eb8305755d90da18d5b9f2R142-R143

In this PR, I am proposing that we just use the append time of the first batch. In this PR https://issues.apache.org/jira/browse/KAFKA-13020, we will read the first batch which will be a control batch, deserialize the value and report that timestamp.